src/services/authorizedItem.service.getPropertiesForItem.test.ts

```diff
@@ -0,0 +1,500 @@
+import { MemberCannotAccess, MemberCannotAdminItem, MemberCannotWriteItem } from '../utils/errors';
+import { AuthorizedItemService } from './authorizedItem.service';
+import { ItemRepository } from './item/item.repository';
+import { ItemVisibilityRepository } from './item/plugins/itemVisibility/itemVisibility.repository';
+import { ItemMembershipRepository } from './itemMembership/membership.repository';
+
+const MOCK_DB = {} as unknown as DBConnection;
+
+const MEMBER = { id: 'shared', name: 'shared' };
+const ITEM = ItemFactory({ id: 'item' });
+
+const buildSharedMembership = (permission: PermissionLevelOptions, item: ItemRaw = ITEM) =>
+  ({ account: MEMBER, permission, item }) as ItemMembershipWithItemAndAccount;
+
+const adminMembership = buildSharedMembership(PermissionLevel.Admin);
+const writeMembership = buildSharedMembership(PermissionLevel.Write);
+const readMembership = buildSharedMembership(PermissionLevel.Read);
+
+const hiddenVisibility = ItemVisibilityFactory({ type: ItemVisibilityType.Hidden, item: ITEM });
+const publicVisibility = ItemVisibilityFactory({ type: ItemVisibilityType.Public, item: ITEM });
+
+const itemMembershipRepository = new ItemMembershipRepository();
+const itemVisibilityRepository = new ItemVisibilityRepository();
+const itemRepository = new ItemRepository();
+
+const authorizationService = new AuthorizedItemService(
+  itemMembershipRepository,
+  itemVisibilityRepository,
+  itemRepository,
+);
+
+describe('getPropertiesForItem', () => {
+  afterEach(() => {
+    jest.restoreAllMocks();
+    jest.clearAllMocks();
+  });
+
+  it('Invalid saved membership', async () => {
+    jest.spyOn(itemVisibilityRepository, 'getByItemPath').mockResolvedValue([]);
+
+    jest
+      .spyOn(itemMembershipRepository, 'getInherited')
+      .mockImplementation(
+        async () => ({ permission: 'anything' }) as unknown as ItemMembershipWithItemAndAccount,
+      );
+
+    // any other member shouldn't access
+    await expect(
+      authorizationService.getPropertiesForItem(MOCK_DB, {
+        permission: PermissionLevel.Admin,
+        actor: MEMBER,
+        item: ITEM,
+      }),
+    ).rejects.toBeInstanceOf(Error);
+  });
+
+  describe('private item shared without permission', () => {
+    it.each([
+      { permission: PermissionLevel.Read, rejects: MemberCannotAccess },
+      { permission: PermissionLevel.Write, rejects: MemberCannotAccess },
+      { permission: PermissionLevel.Admin, rejects: MemberCannotAccess },
+    ])('request $permission should throw', async ({ permission, rejects }) => {
+      jest.spyOn(itemVisibilityRepository, 'getByItemPath').mockResolvedValue([]);
+      jest.spyOn(itemMembershipRepository, 'getInherited').mockImplementation(async () => null);
+
+      await expect(
+        authorizationService.getPropertiesForItem(MOCK_DB, {
+          permission,
+          actor: MEMBER,
+          item: ITEM,
+        }),
+      ).rejects.toBeInstanceOf(rejects);
+    });
+  });
```

Comment:
```suggestion
    it.each([
      { permission: PermissionLevel.Read },
      { permission: PermissionLevel.Write },
      { permission: PermissionLevel.Admin },
    ])('request $permission should throw', async ({ permission }) => {
      jest.spyOn(itemVisibilityRepository, 'getByItemPath').mockResolvedValue([]);
      jest.spyOn(itemMembershipRepository, 'getInherited').mockImplementation(async () => null);

      await expect(
        authorizationService.getPropertiesForItem(MOCK_DB, {
          permission,
          actor: MEMBER,
          item: ITEM,
        }),
      ).rejects.toBeInstanceOf(MemberCannotAccess);
    });
  });
```

Reply:
Seeing the rest of the tests I understand that having the rejects allows to expect different errors based on the permission requested. Question is: Should they be different errors, or should we simply add the requested permission in a generic `Requested permission not allowed` error ?